iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/UnixLinkerTool.cpp

```diff
@@ -31,7 +31,43 @@ class UnixLinkerTool : public LinkerTool {
 
   std::string getToolPath() const override {
     auto toolPath = LinkerTool::getToolPath();
-    return toolPath.empty() ? "ld.lld" : toolPath;
+    if (!toolPath.empty()) return toolPath;
+    if (targetTriple.isAndroid()) {
+// TODO(ataei, benvanik): Windows cross-linking android NDK support.
+#if defined(__unix__) && defined(__x86_64__)
+#if defined(__linux__)
+      return llvm::Twine(std::getenv("ANDROID_NDK"))
+          .concat("/toolchains/llvm/prebuilt/linux-x86_64/bin/")
+#elif defined(__APPLE__)
+      return llvm::Twine(std::getenv("ANDROID_NDK"))
+          .concat("/toolchains/llvm/prebuilt/darwin-x86_64/bin/")
+#else
+      return toolPath;
+#endif
+          // TODO(ataei): Set target archicture and ABI from targetTriple.
+          .concat("aarch64-linux-android30-clang++")
+          .str();
+#elif
+      return toolPath
+#endif
+    }
+// TODO(ataei, benvanik): Windows cross-linking discovery support.
+#ifdef __unix__
+#define UNIX_SYS_LINKER_PATH_LENGTH 255
+    auto sysLinkers = {"ld", "ld.gold", "lld.ld"};
+    for (auto syslinker : sysLinkers) {
+      FILE *pipe =
+          popen(llvm::Twine("which ").concat(syslinker).str().c_str(), "r");
+      char linkerPath[UNIX_SYS_LINKER_PATH_LENGTH];
+      if (fgets(linkerPath, sizeof(linkerPath), pipe) != NULL) {
+        return strtok(linkerPath, "\n");
+      }
```

Comment:
I wonder if there's a helper function for this in LLVM...

Reply:
There is `llvm::sys::ExecuteAndWait` but its API is very verbose for just invoking which and read its stdout